### PR TITLE
Update version to 0.8.x

### PR DESCRIPTION
**Pull request by BenjamenMeyer**
- Updating the engine version to 0.8.x
- Adds basic support for #152
- How to change version info is documented at https://github.com/vegastrike/Vega-Strike-Engine-Source/wiki/Changing-Vega-Strike-Engine-Version-Numbers

Thank you for submitting a pull request and becoming a contributor to the Vega Strike Core Engine.

Please answer the following:

Code Changes:
- [x] Have the PR Validation Tests been run? See https://github.com/vegastrike/Vega-Strike-Engine-Source/wiki/Pull-Request-Validation
- [ ] This is a documentation change only

Issues:
- #152 

Purpose:
- What is this pull request trying to do? Update the version for the master branch
- What release is this for? 0.8.x
- Is there a project or milestone we should apply this to? 0.8.x

### CODE REVIEW

**Comment by evertvorster** [comment]
Hmm.. what is this --version supposed to do?
I would have thought it would print out the version of the VegaStrike engine and then exit, but this is what I get at the console:
[evert@Vorster vegastrike-engine-git]$ vegastrike --version
[2020-11-28 09:05:13.682702] [0x00007f2608d2ac00] [trace]   Registering codec ogg
[2020-11-28 09:05:13.682731] [0x00007f2608d2ac00] [trace]   .

 In path /data/OS/aur/vegastrike-engine-git
Vega Strike  
See http://www.gnu.org/copyleft/gpl.html for license details.

ARG #1 = --version
Command line options for vegastrike

 -D -d   Specify data directory
 -N -n   Number of players
 -M -m   Specify a mod to play
 -P -p   Specify player location
 -J -j   Start in a specific system
 -A -a   Normal resolution (800x600)
 -H -h   High resolution (1024x768)
 -V -v   Super high resolution (1280x1024)
 --net   Networking Enabled (Experimental)
 --debug[=#]     Enable debugging output, 1 major warnings, 2 medium, 3 developer notes
 --test-audio    Run audio tests
 --version       Print the version and exit

[evert@Vorster vegastrike-engine-git]$

**Comment by BenjamenMeyer** [comment]
Thanks. I'll look closer. It should work line '--help'.

**Comment by BenjamenMeyer** [comment]
Found my linker issue...must have been a while since I did a master build as I was missing libgtk-3-dev (Ubuntu); fixed that and it built right up. Verified that `--version` actually does print the version now (sorry about that dumb bug).

**Comment by BenjamenMeyer** [comment]
I'll try to play test later on.

File: engine/src/main.cpp [hunk 1]
```diff
@@ -847,6 +851,9 @@ std::string ParseCommandLine( int argc, char **lpCmdLine )
                 } else if (strcmp( lpCmdLine[i], "--help" ) == 0) {
                     cout<<helpmessage;
                     exit( 0 );
+                } else if (strcmp( lpCmdLine[i], "--version" ) == 0) {
+                    cout<<versionmessage;
+                    exit( 0 );
```
Comment (stephengtuggy) [inline engine/src/main.cpp:856]: Cool; looks like this will work now.

**Review by stephengtuggy — CHANGES_REQUESTED** [review]
Can we leave out the `TWEAK` version number? Thanks. Also, there are more places that will need to change; see my additions to https://github.com/vegastrike/Vega-Strike-Engine-Source/wiki/Changing-Vega-Strike-Engine-Version-Numbers .

**Comment by BenjamenMeyer** [comment]
@stephengtuggy I was thinking of using the Tweak part for the git hash at some point. We don't need it in the packaging, but printing it in --version would be useful.

**Comment by stephengtuggy** [comment]
> @stephengtuggy I was thinking of using the Tweak part for the git hash at some point. We don't need it in the packaging, but printing it in --version would be useful.

@BenjamenMeyer Yeah, now that you mention it, I can see the usefulness of doing that. Perhaps we can put the git hash after a plus sign, as in https://semver.org/#spec-item-10 .

**Review by evertvorster — APPROVED** [review]
It works as advertised now. 
All good in my book!

File: engine/CMakeLists.txt [hunk 2]
```diff
@@ -33,8 +33,19 @@
 
 CMAKE_MINIMUM_REQUIRED(VERSION 3.5)
 SET(CMAKE_CXX_STANDARD 11)
+SET(VEGASTRIKE_VERSION_MAJOR "0")
+SET(VEGASTRIKE_VERSION_MINOR "8")
+SET(VEGASTRIKE_VERSION_PATCH "0")
+SET(VEGASTRIKE_VERSION_TWEAK "0")
```
[the remark this answers is not in the document]
Comment (BenjamenMeyer) [reply engine/CMakeLists.txt:39]: As noted below in (https://github.com/vegastrike/Vega-Strike-Engine-Source/pull/347#issuecomment-735465038) I'd like to use that for the Git Hash.

CPack makes no mention of using Tweak; so it should only show up in the code once we automate things. We can format however, if we want to do x.y.z+h that's fine.

[the remark this answers is not in the document]
Comment (BenjamenMeyer) [reply engine/CMakeLists.txt:42]: Added a line of documentation about the intent for that field.

[the remark this answers is not in the document]
Comment (stephengtuggy) [reply engine/CMakeLists.txt:42]: @BenjamenMeyer Cool. Looks good.

File: engine/src/main.cpp [hunk 3]
```diff
@@ -759,7 +759,8 @@ const char helpmessage[] =
     " --version \t Print the version and exit\n"
     "\n";
 const char versionmessage[] =
-    "Vega Strike Engine Version 0.8.0.0\n"
+    // (BenjamenMeyer) this will be `major.minor.patch+githash` once all is said and done
+    "Vega Strike Engine Version 0.8.0+0\n"
```
Comment (stephengtuggy) [inline engine/src/main.cpp:763]: Nice 👍🏼

**Review by stephengtuggy — APPROVED** [review]
Looks good to me!

**Pull request merged**